rpc/types.go

```diff
@@ -170,54 +170,64 @@ type FeeEstimate struct {
 	OverallFee NumAsHex `json:"overall_fee"`
 }
 
-type TransactionState string
+type TxnExecutionStatus string
 
 const (
-	TransactionAcceptedOnL1 TransactionState = "ACCEPTED_ON_L1"
-	TransactionAcceptedOnL2 TransactionState = "ACCEPTED_ON_L2"
-	TransactionNotReceived  TransactionState = "NOT_RECEIVED"
-	TransactionPending      TransactionState = "PENDING"
-	TransactionReceived     TransactionState = "RECEIVED"
-	TransactionRejected     TransactionState = "REJECTED"
+	TxnExecutionStatusSUCCEEDED TxnExecutionStatus = "SUCCEEDED"
+	TxnExecutionStatusREVERTED  TxnExecutionStatus = "REVERTED"
```

Comment:
I believe there is one more valid execution status state: `RECEIVED` based off of https://github.com/starkware-libs/starknet-specs/pull/134 .

I confirmed with Oded (from Starkware) who said: ```so indeed, even if starknet_addInvokeTransaction returns the tx hash, it is possible the tx will not be included in a block, and therefore the finality status will not be "accepted by L1" or "accepted by L2", but it will be "received".
The only case this is possible is if finality status is received and the execution status is rejected```

Reply:
Hey, thanks for catching this. Technically it's not in the spec, but I agree it should probably be added. @cicr99 what are your thoughts?

Reply:
I'm triple-checking with Starknet to make sure I understand it correctly. Still waiting on hearing back but will let you know!

Reply:
You can ignore my comments about adding status states. I got a confirmation from Starkware (Ariel) that

> TLDR is
the pending status no longer exists for a transaction, ACCEPTED_ON_L2 is the “lowest” finality status (we might also add RECEIVED there for 0.5.0)
the only difference between pending receipt and receipt is the inclusion of block related fields (block_hash, block_number), note that a pending receipt doesn’t affect the finality, you know with certainty that it will be included in the next block

and 

> tldr the starknet-specs have not included RECEIVED status yet. So currently in v0.4, will the API just error for a tx hash that hasn't been ACCEPTED_ON_L2 yet

Hope that clears things up